py/objboundmeth.c

```diff
@@ -83,6 +83,21 @@ STATIC mp_obj_t bound_meth_call(mp_obj_t self_in, size_t n_args, size_t n_kw, co
     return mp_call_method_self_n_kw(self->meth, self->self, n_args, n_kw, args);
 }
 
+STATIC mp_obj_t bound_meth_binary_op(mp_binary_op_t op, mp_obj_t lhs_in, mp_obj_t rhs_in) {
+    mp_obj_bound_meth_t *o = MP_OBJ_TO_PTR(lhs_in);
+    switch (op) {
+        case MP_BINARY_OP_EQUAL: {
+            mp_obj_bound_meth_t *rhs = MP_OBJ_TO_PTR(rhs_in);
```

Comment:
What if rhs_in is a different kind of object?

Reply:
I tried this but ran into issues with it. In my testing, it never seems to be called with another object, but I am not certain if this is always the case.

However, as I am writing I thought of something that could solve those issues, I will give a go tomorrow.

Reply:
I added a mp_obj_is_type check to make sure it is the same type. 

```c
if (!mp_obj_is_type(rhs_in, &mp_type_bound_meth)) {
    return mp_const_false;
}
```